sources/packages/backend/apps/api/src/services/notification/notification.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from "@nestjs/common";
+import {
+  RecordDataModelService,
+  Notification,
+  User,
+  NotificationMessage,
+  NotificationMessageType,
+} from "@sims/sims-db";
+import { DataSource, UpdateResult } from "typeorm";
+
+@Injectable()
+export class NotificationService extends RecordDataModelService<Notification> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(Notification));
+  }
+
+  /**
+   * Creates the inbox notification record.
+   * @param userId id of the user who will receive the message.
+   * @param messageType message type of the notification.
+   * @param templateId template id of the notification.
+   * @param auditUserId id of the user creating the notification.
+   * @param messagePayload notification payload.
+   * @returns payload of the record created.
+   */
+  async saveNotification(
+    userId: number,
+    messageType: NotificationMessageType,
+    templateId: string,
+    messagePayload: unknown,
+    auditUserId: number,
+  ): Promise<Notification> {
+    const notification = new Notification();
+    notification.templateId = templateId;
+    notification.user = { id: userId } as User;
+    notification.creator = { id: auditUserId } as User;
+    notification.messagePayload = messagePayload;
+    notification.createdAt = new Date();
```

Comment:
We do not need to explicitly set it unless we would like to keep it in sync with other operations, which is not the case here. Please remove it.